client/apikey/apikey.go

```diff
@@ -1,123 +1,87 @@
 package apikey
 
 import (
-	"bytes"
 	"encoding/json"
-	"fmt"
 	"github.com/banyansecurity/terraform-banyan-provider/client/restclient"
 	"github.com/pkg/errors"
 	"io/ioutil"
-	"log"
-	"net/http"
 	"net/url"
 )
 
 type ApiKey struct {
-	restClient *restclient.RestClient
+	restClient *restclient.Client
 }
 
-// NewClient returns a new client for interacting with the apikey resource
-func NewClient(restClient *restclient.RestClient) Clienter {
+const apiVersion = "api/v2"
+const component = "api_key"
+
+func NewClient(restClient *restclient.Client) Client {
 	apikeyClient := ApiKey{
 		restClient: restClient,
 	}
 	return &apikeyClient
 }
 
-// Clienter is used for performing CRUD operations on the apikey resource
-type Clienter interface {
+type Client interface {
 	Get(id string) (apikey Data, err error)
 	Create(post Post) (createdApiKey Data, err error)
-	Update(post Post) (updatedApiKey Data, err error)
+	Update(id string, post Post) (updatedApiKey Data, err error)
 	Delete(id string) (err error)
 }
 
 func (k *ApiKey) Get(id string) (apikey Data, err error) {
-	if id == "" {
-		err = errors.New("need an id to get a apikey")
+	resp, err := k.restClient.Read(apiVersion, component, id, "")
+	if err != nil {
```

Comment:
Should we also be checking `resp` for error conditions?

Something like:

```go
if resp.StatusCode < 200 || resp.StatusCode >= 300 {
  return fmt.Errorf("got failing status: %d, response: %s", resp.StatusCode, body)
}
```

Reply:
ive added this to to restClient.Read